apps/wakunode2/app.nim

```diff
@@ -222,6 +226,41 @@ proc setupWakuArchiveDriver(dbUrl: string, vacuum: bool, migrate: bool): AppResu
     let driver = QueueDriver.new()  # Defaults to a capacity of 25.000 messages
     ok(driver)
 
+proc setupWakuArchive*(app: var App): AppResult[void] =
+  ## Waku archive
+
+  if not app.conf.store:
+    return ok()
+
+  # Message storage
+  let dbUrlValidationRes = validateDbUrl(app.conf.storeMessageDbUrl)
+  if dbUrlValidationRes.isErr():
+    return err("failed to configure the message store database connection" & dbUrlValidationRes.error)
+
+  let archiveDriverRes = setupWakuArchiveDriver(dbUrlValidationRes.get(),
+                                                vacuum = app.conf.storeMessageDbVacuum,
+                                                migrate = app.conf.storeMessageDbMigration)
+  if archiveDriverRes.isOk():
+    app.archiveDriver = some(archiveDriverRes.get())
+  else:
+    return err("failed to configure archive driver" & archiveDriverRes.error)
+
+  # Message store retention policy
+  let storeMessageRetentionPolicyRes = validateStoreMessageRetentionPolicy(app.conf.storeMessageRetentionPolicy)
+  if storeMessageRetentionPolicyRes.isErr():
+    return err("failed to configure the message retention policy" & storeMessageRetentionPolicyRes.error)
```

Comment:
The PR LGTM!
Shall we add a separator (e.g. colon or `" [" & error & "]"`) ?
There few other cases in this file that also need to separate errors.

Reply:
Added. Note that these errors are here for backward compatibility and temporarily until the rest of the `App` is reworked and refactored.